core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -278,7 +280,26 @@ class DAGSchedulerSuite extends SparkFunSuite with LocalSparkContext with TimeLi
 
   override def beforeEach(): Unit = {
     super.beforeEach()
-    init(new SparkConf())
+  }
+
+  override protected def test(testName: String, testTags: Tag*)(testFun: => Any)
+      (implicit pos: Position): Unit = {
+    testWithSparkConf(testName, testTags: _*)()(testFun)(pos)
+  }
+
+  private def testWithSparkConf(testName: String, testTags: Tag*)
+      (pairs: (String, String)*)(testFun: => Any)(implicit pos: Position): Unit = {
+    super.test(testName, testTags: _*) {
+      withSparkConf(pairs: _*)(testFun)
+    }
+  }
+
+  /** Sets all configurations specified in `pairs`, calls `init`, and then calls `testFun` */
+  private def withSparkConf(pairs: (String, String)*)(testFun: => Any): Unit = {
```

Comment:
For `test()`, `withSparkConf()`, shall we extract them into a base class? I guess they could be used by other test suites as well?

Reply:
OK. I will simulate `SQLHelper`.